### PR TITLE
[Merged by Bors] - refactor(GroupTheory/Sylow): use namespacing and dot-notation

### PR DESCRIPTION
**Pull request by tb65536**
This PR cleans up `Sylow.lean` by using namespacing and dot-notation.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/GroupTheory/Sylow.lean [hunk 1]
```diff
@@ -129,28 +125,28 @@ def comapOfKerIsPGroup (hϕ : IsPGroup p ϕ.ker) (h : ↑P ≤ ϕ.range) : Sylow
       exact (comap_map_eq_self ((P.1.ker_le_comap ϕ).trans hle)).symm }
 
 @[simp]
-theorem coe_comapOfKerIsPGroup (hϕ : IsPGroup p ϕ.ker) (h : ↑P ≤ ϕ.range) :
-    (P.comapOfKerIsPGroup ϕ hϕ h : Subgroup K) = Subgroup.comap ϕ ↑P :=
+theorem coe_comapOfKerIsPGroup (hϕ : IsPGroup p ϕ.ker) (h : P ≤ ϕ.range) :
+    P.comapOfKerIsPGroup ϕ hϕ h = P.comap ϕ :=
```
Comment (grunweg) [inline Mathlib/GroupTheory/Sylow.lean:129]: Flagging for reviewers: there are a few instances of removed type ascriptions. I have not thought closely if these need further attention or not.

File: Mathlib/GroupTheory/Sylow.lean [hunk 2]
```diff
@@ -108,19 +108,15 @@ def ofCard [Finite G] {p : ℕ} [Fact p.Prime] (H : Subgroup G)
...
 @[simp, norm_cast]
 theorem coe_ofCard [Finite G] {p : ℕ} [Fact p.Prime] (H : Subgroup G)
-    (card_eq : Nat.card H = p ^ (Nat.card G).factorization p) : ↑(ofCard H card_eq) = H :=
+    (card_eq : Nat.card H = p ^ (Nat.card G).factorization p) : ofCard H card_eq = H :=
   rfl
 
 variable (P : Sylow p G)
 
-/-- The action by a Sylow subgroup is the action by the underlying group. -/
-instance mulActionLeft {α : Type*} [MulAction G α] : MulAction P α :=
```
Comment (grunweg) [inline Mathlib/GroupTheory/Sylow.lean:117]: Just checking: I presume this instance is not needed any more?

File: Mathlib/GroupTheory/Sylow.lean (hunk 1 above)
Comment (grunweg) [reply Mathlib/GroupTheory/Sylow.lean:129]: Similarly for removed coercions.

**Review by grunweg — COMMENTED** [review]
Three minor questions (which another reviewer or maintainer could take a look at). Rest looks good to me. Thanks for doing this!
maintainer merge

**Comment by jcommelin** [comment]
Thanks :tada:

bors merge

**Pull request closed without merging**